internal/util/iterator/values.go

```diff
@@ -85,17 +85,21 @@ func ConsumeValuesN[K, V any](iter Interface[K, V], n int) ([]V, error) {
 }
 
 // Values returns an iterator over values of another iterator.
+// It will be added to the given closer.
 //
 // Close method closes the underlying iterator.
 // For that reason, there is no need to track both iterators.
-func Values[K, V any](iter Interface[K, V]) Interface[struct{}, V] {
+func Values[K, V any](iter Interface[K, V], closer *MultiCloser) Interface[struct{}, V] {
```

Comment:
Let's revert that for now. Let's keep MultiCloser for handlers only

Reply:
Okie, I've reverted. I noticed that not all stages handled closer upon creating iterator and I thought adding that would make it easier.

Reply:
I'm not completely sure about that yet, but I think we will change the behavior of iterator's Close method working "back" towards the begging on the chain by closing the underlying iterator. If we do that, then iterator's Close and MultiCloser will be clearly separated, and we could add it to all iterators